test/integration/event_log_test.rb

```diff
@@ -237,6 +237,34 @@ class EventLogIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_content?("You do not have permission to perform this action")
   end
 
+  test "record user's login ip address" do
+    visit root_path
+    signin_with(@user)
+
+    ip_address = @user.event_logs.last.trailing_message.split.first
+    assert ip_address.match(/\A[[:alnum:].:]+$/)
+  end
+
+  test "record user's login browser" do
+    visit root_path
+    signin_with(@user)
+
+    browser = @user.event_logs.last.trailing_message.split.last
+    assert browser.match(/\A[[:alpha:]]+/)
+  end
```

Comment:
As above, it'd be good to assert against an actual browser value here instead of just saying the 2nd part of the trailing message is all word characters.